MAISTRA-1898: Update to latest upstream 1.6

jwendell opened the pull request.

istio-testing committed:
[release-1.6] Add version label to `pilot_xds`  metric (#25284)

* Add version label to metric

Signed-off-by: Liam White <liam@tetrate.io>

* back to gauge

Signed-off-by: Liam White <liam@tetrate.io>

Co-authored-by: Liam White <liam@tetrate.io>

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#25426)

elfinhe committed:
Never execute waitUpgradeComplete and retrieveControlPlaneVersion when upgrade with dry run option (#25207) (#25425)

Co-authored-by: Zufar Dhiyaulhaq <zufardhiyaulhaq@gmail.com>

istio-testing committed:
Fix egress gateway config. (#25514)

Co-authored-by: Oliver Liu <yonggangl@google.com>

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#25553)

istio-testing committed:
[release-1.6] Potential bug in tests.mk file (#25558)

* fixed typo

* fix the default behavior

Co-authored-by: zhengzhey <zhengzhey@google.com>

istio-testing committed:
Remove potentionally massive log line (#25610)

For clusters with many gateways this is a massive line, when its not
very useful

Co-authored-by: John Howard <howardjohn@google.com>

howardjohn committed:
1.6: Trigger endpoint update when pod comes in (#25161) (#25579)

* 1.6: Trigger endpoint update when pod comes in (#25161)

* change log

* change debug log

istio-testing committed:
Optimize ResolveGatewayName (#25611)

In large clusters this has been shown to uses ~20% of CPU and ~10% of
allocations

BenchmarkResolveGatewayName/old-8               10342688
1222 ns/op             320 B/op         16 allocs/op
BenchmarkResolveGatewayName/new-8               39173893
322 ns/op              64 B/op          4 allocs/op

Co-authored-by: John Howard <howardjohn@google.com>

howardjohn committed:
1.6: pilot: fix n^2 scaling behavior with gateways (#25613)

Cherrypick of https://github.com/istio/istio/pull/25136/ to 1.6

ramaraochavali committed:
disable enforcing successrate based outlier detection (#25567)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

morvencao committed:
update proxy readiness probe to port 15021 and cleanup demo profile. (#25635)

Commit:
Update the base image versions to latest for bookinfo sample (#24967) (#25645)

* Update the base image versions to latest for bookinfo sample

* use IPv4 address instead of IPv6 address

stewartbutler committed:
Changing CNI images from latest to 1.6-dev tag (#25811)

istio-testing committed:
Fix tombstone recovery bug in Galley (#25413)

Co-authored-by: Brad Ison <brad.ison@redhat.com>

GregHanson committed:
maunal cherry pick of 3b58e4c5e #25588 (#25751)

jacob-delgado committed:
[release-1.6] Update 1.6 dependencies (#25817)

* Update 1.6 dependencies

UPDATE_BRANCH=release-1.6 ./bin/update_deps.sh; make gen

* Fix integration test

istio-testing committed:
[release-1.6] verify if namespace exists during bookinfo cleanup (#25831)

* verify if namespace exists during bookinfo cleanup

* fix lint

Co-authored-by: shamsher31 <shaansar@redhat.com>

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#25839)

hzxuzhonghu committed:
Do not enable auto mtls when cluster type is `Cluster_ORIGINAL_DST` (#24319) (#25829)

* disable tls for headless service when mtls is auto detected

* move it to conditionallyConvertToIstioMtls

* cleanup

* fix e2e

howardjohn committed:
Bump base image for 1.6 (#25845)

howardjohn committed:
Fix regression for Endpoints without pod reference (#25978) (#25987)

(cherry picked from commit d5ab2ebfa13107099a6fed596b5201f88ad28d24)
(cherry picked from commit 805c4032fae7c3b81b3dcc359eba0d9b7b7dc2b8)

jacob-delgado committed:
[release-1.6] CVE fix for 1.6.8 (#26371)

* fix authz suffix matching in TCP (#29)

* update the tests (#31)

Co-authored-by: Yangmin Zhu <ymzhu@google.com>

istio-testing committed:
Expand endpoints before pod test to check pod (#26032)

This ensures we are actually getting the right pod, and populating the
correct service account information. It doesn't fix any bug - the code
works today, just expanding the testing

Co-authored-by: John Howard <howardjohn@google.com>

istio-testing committed:
remove istio-validation container when running istioctl rm (#26189)

Co-authored-by: Tariq Ibrahim <tariq181290@gmail.com>

bianpengyuan committed:
[release-1.6] Increase default proto sniffing timeout to 5s (#25954)

* increase default proto sniffing timeout to 5s

* make gen

istio-testing committed:
[release-1.6] properly drain gateway listeners (#26068)

* drain all listeners for gateway

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Co-authored-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Fix regression in gateway name resolution (#26352)

Fixes https://github.com/istio/istio/issues/26264

Co-authored-by: John Howard <howardjohn@google.com>

istio-testing committed:
Fix inaccurate endpointsPendingPodUpdate metric (#25906)

This currently will be outdate when an update comes in, and is only
updated when the error is retriggered

Co-authored-by: John Howard <howardjohn@google.com>

istio-testing committed:
Fix networking.HTTPMatchRequest.WithoutHeaders conflict detect (#26064)

Co-authored-by: xuzhonghu <xuzhonghu@huawei.com>

gargnupur committed:
Change to comma separated value for app_container (#25441) (#25864)

* Change to comma separated value for app_container

Signed-off-by: gargnupur <gargnupur@google.com>

Run make gen

Signed-off-by: gargnupur <gargnupur@google.com>

Add test for container  name

Signed-off-by: gargnupur <gargnupur@google.com>

Update VM test files

Signed-off-by: gargnupur <gargnupur@google.com>

Change to comma separated value for app_container

Signed-off-by: gargnupur <gargnupur@google.com>

Run make gen

Signed-off-by: gargnupur <gargnupur@google.com>

Add test for container  name

Signed-off-by: gargnupur <gargnupur@google.com>

* Fix vm test

Signed-off-by: gargnupur <gargnupur@google.com>

nschhina committed:
[Release 1.6] Manual cherrypick of #25755 (#26122)

* manual cherrypick

* fix typo

morvencao committed:
cherrypick 24696. (#26348)

williamaronli committed:
[Bug fix][Version 1.6]Fix ingress SDS not getting secret updates and create unit tests (#25681)

* Fix ingress SDS not getting secret updates and create unit tests (#24817)

* fix ingress SDS not getting secret updates issue and add unit test

* change to use release

* address comments

* fix lint

* add more comment and delete useless code

(cherry picked from commit 120c9b053de0ddea3cfd1830363491d39217e200)

* delete version conflict issue

samjo-nyang committed:
Add missing telemetry.loadshedding.* options to mixer container args (#25867)

istio-testing committed:
Fix egressgateway ports (#26460)

Cannot bind to port 80/443 since we run as non root by default

Co-authored-by: John Howard <howardjohn@google.com>

Commit:
Update version of x/text package (#26464)

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#26474)

williamaronli committed:
cherry-pick Remove SDS Timeout for default and root case (#26533)

hzxuzhonghu committed:
[release-1.6] Fix headless svc instances scale  (#26636) (#26681)

* Fix headless svc instances scale  (#26636)

* Fix configupdate for service

* Add unit test

* add release-note

* rm

istio-testing committed:
feat:istioctl x add-to-mesh and remove-from-mesh Should not affect OwnerReferences (#26770)

Co-authored-by: tanjunchen <tanjunchen20@gmail.com>

stevenctl committed:
[release-1.6] allow specifying network for cluster without meshNetworks being configured (#26657)

* allow specifying network for cluster without meshNetworks being configured (#26618)

* allow specifying network without meshNetworks fully configured

* remove redundant slice alloc and add safety check for clusterID

* move cluster id check

* set clustername to match in tests

* isControllerForProxy

(cherry picked from commit 97c21c4e6036a3e09484ae1d5c007a94c134c2ff)

* fix lint

istio-testing committed:
[release-1.6] cache readiness state with TTL (#26742)

* remove stats filter in readiness probe

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* cache readiness state with a TTL

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* revert the parse state change

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* rename variable

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* add tests

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* make readiness timeout configurable

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* continuously check for readiness on failure

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* lint

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Co-authored-by: Rama Chavali <rama.rao@salesforce.com>

Commit:
trust domain validation cherry pick for release-1.6 (#26660)

* Enable trust domain validation in transport socket level (#26224)

* feature

* make gen

revert false

fix unit test

add aliases

fix lint

break line

revert duplicate test

only do this for sidecar

fix

alias test

dont export constructSAN

fix comments

fix bug

change

lint

create util function

add copyright

make gen

* remove dup

* another dup

* fix tests

* rename func

* lint line too long

* add todos

* add release note

* make copy

* explicit true

* fix issue

* fix unit test

* make gen

* update release note

* remove dead code

* remove cluster.local

* fix unit test

Co-authored-by: Navraj Singh Chhina <navuchhina@live.com>

yangminzhu committed:
release-1.6: add e2e tests for trust domain validation (#26804)

williamaronli committed:
add outputOutputKeyCertToDir to enable workload load from the file cert (#26897)

Change-Id: I447dc3a7be70ad1086e30fdd7edbe0e31507bf3a

istio-testing committed:
Fix ledger capacity size (#26670)

10^5 == 15 (xor) not 10 to the 5th power

Co-authored-by: John Howard <howardjohn@google.com>

howardjohn committed:
Bump 1.6 image (#26934)

stevenctl committed:
[release-1.6] manual cherry-pick of #26899 (#27002)

hzxuzhonghu committed:
Fix bug: service occur later than endpoint (#26986)

istio-testing committed:
add update permission to servicemonitor. (#26996)

Co-authored-by: morvencao <morvencao@gmail.com>

hzxuzhonghu committed:
Fix eds: gateways missing endpoint instances of headless service (#27121)

gargnupur committed:
Add quotes in log sampling config and add it in the stackdriver test (#27007) (#27214)

Signed-off-by: gargnupur <gargnupur@google.com>

istio-testing committed:
cache envoy readiness value (#27269)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

Co-authored-by: Rama Chavali <rama.rao@salesforce.com>

bianpengyuan committed:
Trim job suffix to extract out cron job name for workload metadata (#… (#27252)

* Trim job suffix to extract out cron job name for workload metadata (#27195)

* special case cron job processing in webhook

* update

* fix test

richardwxn committed:
remove deprecated format from help message (#27220)

ramaraochavali committed:
[release-1.6] do not apply locality load balancer settings for inbound clusters (#27353)

* manual cherrypick of 27295

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

* fix compilation

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

istio-testing committed:
Fix endpointShardz leak (#27451)

Fixes https://github.com/istio/istio/issues/24927

Current we pass the service *name* to ServiceUpdate. In the SE registry,
we pass hostname, as expected. The result of using the name is that on
service deletion, we actually use the wrong key for deletion.

Co-authored-by: John Howard <howardjohn@google.com>

immortanjoe committed:
[release-1.6] fix error service name in Cleanup (#27461)

* fix error service name in Cleanup

* Fix endpointShardz leak (#24947)

Fixes https://github.com/istio/istio/issues/24927

Current we pass the service *name* to ServiceUpdate. In the SE registry,
we pass hostname, as expected. The result of using the name is that on
service deletion, we actually use the wrong key for deletion.

Co-authored-by: John Howard <howardjohn@google.com>

morvencao committed:
cherrypick 26456 to release-1.6 (#27539)

* cherrypick 26456 to release-1.6

* fix lint.

aattuluri committed:
1.6 patch for issue #27427 (#27470)

* Add option to skip multicluster envoy filter (#27427)

* Add a flag to include envoy filter and make global domain configurable

* Fix values.

* Update schema

* Add release notes

* Update generated files.

* Trim global domain suffix

* Fix typo

* Add right values for multicluster

* Re generate the values_types proto.

howardjohn committed:
[1.6] Stop creating invalid double wildcard domains (#27362) (#27528)

* Stop creating invalid double wildcard domains (#27362)

* Stop creating invalid double wildcard domains

For https://github.com/istio/istio/issues/25350

Does not fully fix the problem, see envoy issue

* Add release notes

(cherry picked from commit 0e8aebca29dc1dd0df1af01e5701176f39fb7d43)

* Update pilot/pkg/networking/core/v1alpha3/gateway.go

Co-authored-by: Shamsher Ansari <shaansar@redhat.com>

Co-authored-by: Shamsher Ansari <shaansar@redhat.com>

istio-testing committed:
Automator: update common-files@release-1.6 in istio/istio@release-1.6 (#27733)

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#27736)

brian-avery committed:
Update deps (#27754)

howardjohn committed:
Bump 1.6 base image (#27812)

xulingqing committed:
[release-1.6]Cherry-pick: remove cacert.pem and use x509.SystemCertPool() (#27766)

* [release-1.6] cherry pick: remove cacert.pem and use x509.SystemCertPool() (#27657)

* remove tests/integration/mixer/helper.go cacert.pem dependency

* add release note

hzxuzhonghu committed:
manually cherry pick #27864 (#27985)

howardjohn committed:
Bump base image (#28118)

howardjohn committed:
Conditionally keep inbound listeners (#28111)

jacob-delgado committed:
Update fake-stackdriver to the one used in istio/proxy release-1.6 (#28152)

See
https://github.com/istio/proxy/blob/release-1.6/test/envoye2e/stackdriver_plugin/cmd/Makefile#L21

howardjohn committed:
optimize memory usage (#25531) (#25532) (#28369)

(cherry picked from commit ec9166a1e46945c6e301b1165c7ec1cc61b0aeb6)

Co-authored-by: Dozer <mail@dozer.cc>

howardjohn committed:
[1.6] Adjust inbound and outbound filter chains (#28442) (#28554)

* Adjust inbound and outbound filter chains (#28442)

* Adjust inbound and outbound filter chains

* fix test

(cherry picked from commit 1f422dd56161d16f9e77bcde5ccc7cd3e065e870)

* fixes

jacob-delgado committed:
Update release-1.6 dependencies (#28622)

* Run UPDATE_BRANCH=release-1.6 ./bin/update_deps.sh; make gen

* Fix broken telemetry test

istio-testing committed:
Automator: update common-files@release-1.6 in istio/istio@release-1.6 (#28868)

istio-testing committed:
Automator: update istio/api@release-1.6 dependency in istio/istio@release-1.6 (#28870)

Commit:
Update 1.6 base image (#28923)

sel-vcc committed:
[release-1.6] cherry-pick: do not update root certs for destination rule certs (#27268) (#28869)

* do not update root certs for destination rule certs (#27268)

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>
(cherry picked from commit 2cbbf53da0acb42d23decabd2dacd7929961f515)

* common/scripts: Fix quoting

Quoting was causing CI lint check to fail:
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/28869/lint_istio_release-1.6/1328742716099530752/build-log.txt

Signed-off-by: Steve Larkin <steve.larkin+vcc@gmail.com>

Co-authored-by: Rama Chavali <rama.rao@salesforce.com>

hzxuzhonghu committed:
Workloadentry add should trigger push to the associated proxy (#28846)

jacob-delgado committed:
[release-1.6] Cherry pick: fix conflict between replicas with hpa (#29051)

* fix conflict between replicas with hpa.

* add release note.

* Update releasenotes/notes/28916.yaml

Co-authored-by: Shamsher Ansari <shaansar@redhat.com>

Co-authored-by: morvencao <morvencao@gmail.com>
Co-authored-by: Morven Cao <llcao@cn.ibm.com>
Co-authored-by: Shamsher Ansari <shaansar@redhat.com>

jacob-delgado committed:
[release-1/6] update dependencies (#29086)

* run UPDATE_BRANCH=release-1.6 ./bin/update_deps.sh; make gen

* Revert quote removal

Commit:
Update base image for build (#29094)

jwendell committed:
Merge remote-tracking branch 'upstream/release-1.6' into MAISTRA-1898

jwendell committed:
Merge branch 'maistra-2.0' into MAISTRA-1898

jwendell committed:
Update vendor

jwendell commented on pilot/pkg/serviceregistry/kube/controller/controller.go:
```diff
@@ -301,6 +309,7 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 
 	c.services = cache.NewSharedIndexInformer(svcMlw, &v1.Service{}, options.ResyncPeriod,
 		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	c.serviceLister = listerv1.NewServiceLister(c.services.GetIndexer())
```
@dgn @bison ^

bison replied:
If this is just in reference to the lister, it looks fine. As long as the lister is based on an informer using the MLW, it should work like we expect it to.

jwendell commented on pilot/pkg/serviceregistry/kube/controller/controller.go:
```diff
@@ -330,7 +339,16 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 	}
 
-	c.pods = newPodCache(c, mrc, options)
+	c.pods = newPodCache(c, mrc, options, func(key string) {
```
@dgn @bison

bison replied:
Looks fine to me.

jwendell commented:
@dgn @bison I've tagged you both in the non-trivial conflicts I had to fix. Please take a look and let me know if it's wrong.

jwendell committed:
Fix a missing import

jwendell committed:
make gen

jwendell committed:
Merge branch 'maistra-2.0' into MAISTRA-1898

jwendell committed:
Fix merge logic and conflicts in push_context.go

dgn commented:
I'm working on fixing the tests

jwendell commented:
@dgn ok, I was about to start this task. Thanks.

dgn committed:
MAISTRA-1898 Fix mTLS unit tests

The update to the latest 1.6 introduced a new test case which needed
a small change to make it compatible with our TLS parameter changes

dgn commented:
Okay, I pushed a commit that should fix the unit test failures. The integration test failure looks like a flake to me (it's a test that relies on `sleep()` to validate cert lifetimes...)

bison reviewed (approved):
Obviously mostly skimmed this... but if the tests pass, then LGTM.
